udata/search/commands.py

```diff
@@ -44,14 +64,41 @@ def index_model(index_name, adapter):
         qs = qs.visible()
     if adapter.exclude_fields:
         qs = qs.exclude(*adapter.exclude_fields)
-    for obj in qs.no_dereference().timeout(False):
-        if adapter.is_indexable(obj):
-            try:
-                doc = adapter.from_model(obj)
-                doc.save(using=es.client, index=index_name)
-            except:
-                log.exception('Unable to index %s "%s"',
-                              model.__name__, str(obj.id))
+
+    docs = iter_qs(qs, adapter)
+    docs = iter_for_index(docs, index_name)
+
+    for ok, info in streaming_bulk(es.client, docs, raise_on_error=False):
+        if not ok:
+            log.error('Unable to index %s "%s": %s', model.__name__,
+                      info['index']['_id'], info['index']['error'])
+
+
+def disable_refresh(index_name):
+    '''
+    Disable refresh to optimize indexing
+
+    See: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-update-settings.html#bulk
+    '''  # noqa
+    es.indices.put_settings(index=index_name, body={
+        'index': {
+            'refresh_interval': '-1'
+        }
+    })
+
+
+def enable_refresh(index_name):
```

Comment:
Maybe put `refresh_interval` as a parameter with a default value of `1s`? In case this needs to be changed/configured.

Reply:
Done :+1: